TM1py/Services/CellService.py

```diff
@@ -950,25 +1019,53 @@ def extract_cellset_cellcount(self, cellset_id: str, **kwargs) -> int:
         response = self._rest.GET(url, **kwargs)
         return int(response.content)
 
-    @tidy_cellset
-    def extract_cellset_csv(self, cellset_id: str, **kwargs) -> str:
+    def extract_cellset_csv(
+            self,
+            cellset_id: str,
+            skip_zeros: bool = True,
+            skip_consolidated_cells: bool = False,
+            skip_rule_derived_cells: bool = False,
+            line_separator: str = "\r\n",
+            **kwargs) -> str:
         """ Execute cellset and return only the 'Content', in csv format
 
         :param cellset_id: String; ID of existing cellset
+        :param skip_zeros: skip zeros in cellset
+        :param skip_consolidated_cells: skip consolidated cells in cellset
+        :param skip_rule_derived_cells: skip rule derived cells in cellset
+        :param line_separator:
         :return: Raw format from TM1.
         """
-        url = "/api/v1/Cellsets('{}')/Content".format(cellset_id)
-        data = self._rest.GET(url, **kwargs)
-        return data.text
+        _, _, rows, columns = self.extract_cellset_composition(cellset_id, delete_cellset=False, **kwargs)
 
-    def extract_cellset_dataframe(self, cellset_id: str, **kwargs) -> pd.DataFrame:
+        cellset_dict = self.extract_cellset_raw(cellset_id, cell_properties=["Value"], skip_contexts=True,
+                                                skip_zeros=skip_zeros, skip_consolidated_cells=skip_consolidated_cells,
+                                                skip_rule_derived_cells=skip_rule_derived_cells,
+                                                delete_cellset=True, **kwargs)
+        return build_csv_from_cellset_dict(rows, columns, cellset_dict, line_separator=line_separator)
+
+    def extract_cellset_dataframe(
+            self,
+            cellset_id: str,
+            skip_zeros: bool = True,
+            skip_consolidated_cells: bool = False,
+            skip_rule_derived_cells: bool = False,
+            **kwargs) -> pd.DataFrame:
         """ Build pandas data frame from cellset_id
 
         :param cellset_id:
+        :param skip_contexts:
+        :param skip_zeros: skip zeros in cellset
+        :param skip_consolidated_cells: skip consolidated cells in cellset
+        :param skip_rule_derived_cells: skip rule derived cells in cellset
+
         :param kwargs:
         :return:
         """
-        raw_csv = self.extract_cellset_csv(cellset_id=cellset_id, delete_cellset=True, **kwargs)
+        raw_csv = self.extract_cellset_csv(cellset_id=cellset_id, skip_zeros=skip_zeros,
+                                           skip_rule_derived_cells=skip_rule_derived_cells,
+                                           skip_consolidated_cells=skip_consolidated_cells,
+                                           **kwargs)
         if not raw_csv:
```

Comment:
If the `cellset` has no data, then the `raw_csv` string has header part in it.. Ideally it will not be empty and will not get into the empty `Dataframe` part. So, lets add a `new line` check to identify existence of data lines. 

`if "\n" not in raw_csv:`

Reply:
Thanks for the catch! 

Actually, in order to be fully backwards compatible, we should return an empty string if the cellset is completely empty, just like the `/Content` did.
I will adjust the `extract_cellset_csv method` and add a test.

Reply:
Yeah, sounds good!

Reply:
I think it would be nicer to add the  `skip` Default arguments to `execute_view_dataframe `and `execute_mdx_dataframe` methods as well , for now `Kwargs` works though.

Reply:
Agree. Added all arguments explicitly. Including `top` and `skip`.